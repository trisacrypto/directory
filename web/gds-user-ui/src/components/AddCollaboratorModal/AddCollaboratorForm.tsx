import { FC, useEffect } from 'react';
import { useForm } from 'react-hook-form';
import { ADD_COLLABORATOR_FORM_METHOD } from './AddCollaboratorFormValidation';
import type { NewCollaborator } from './AddCollaboratorType';
import { useCreateCollaborator } from 'hooks/useCreateCollaborator';
import { Button, Stack, Text, VStack, chakra, useToast } from '@chakra-ui/react';
import { DevTool } from '@hookform/devtools';
import { isProdEnv } from 'application/config';
import { Trans } from '@lingui/react';
import InputFormControl from 'components/ui/InputFormControl';
import CustomToast from 'components/ui/CustomToast';
import CheckboxFormControl from 'components/ui/CheckboxFormControl';
import { t } from '@lingui/macro';
import { useFetchCollaborators } from 'components/Collaborators/useFetchCollaborator';
type Props = {
  onCloseModal: () => void;
};

const AddCollaboratorForm: FC<Props> = (props) => {
  const { onCloseModal } = props;
  const mutation = useCreateCollaborator();
  const toast = useToast();
  const { getAllCollaborators } = useFetchCollaborators();

  const {
    createCollaborator,
    isCreating,
    wasCollaboratorCreated,
    hasCollaboratorFailed,
    errorMessage
  } = mutation;
  const {
    register,
    control,
    handleSubmit,
    getValues: formValues,
    formState: { errors }
  } = useForm<NewCollaborator>(ADD_COLLABORATOR_FORM_METHOD) as any;

  useEffect(() => {
    if (wasCollaboratorCreated) {
      getAllCollaborators();
    }
  }, [wasCollaboratorCreated, getAllCollaborators]);

  useEffect(() => {
    if (wasCollaboratorCreated) {
      onCloseModal();
      toast({
        position: 'top-right',
        title: t`Collaborator has been added successfully`,
        isClosable: true,
        duration: 9000,
        status: 'success'
      });
    }
  }, [wasCollaboratorCreated, onCloseModal, toast]);

  const onSubmit = (data: NewCollaborator) => {
    createCollaborator(data);
    // set collaborators in global state
  };

  if (hasCollaboratorFailed) {
    return (
      <Stack data-test="collaborator-failed">
        <CustomToast
          title="Fail to add a new collaborator"
          description={errorMessage}
          status="error"
        />
      </Stack>
    );
  }

  return (
    <chakra.form onSubmit={handleSubmit(onSubmit)}>
      <VStack>
        <Text>
          <Trans id="Please provide the name of the VASP and email address for the new contact.">
            Please provide the name of the VASP and email address for the new contact.
          </Trans>
        </Text>
        <Text>
          <Trans id="The contact will receive an email to create a TRISA Global Directory Service Account. The invitation to join is valid for 7 calendar days. The contact will be added as a member for the VASP. The contact will have the ability to contribute to certificate requests, check on the status of certificate requests, and complete other actions related to the organization’s TRISA membership.">
            The contact will receive an email to create a TRISA Global Directory Service Account.
            The invitation to join is valid for 7 calendar days. The contact will be added as a
            member for the VASP. The contact will have the ability to contribute to certificate
            requests, check on the status of certificate requests, and complete other actions
            related to the organization’s TRISA membership.
          </Trans>
        </Text>
      </VStack>

      <CheckboxFormControl
        controlId="agreed"
        mt={2}
        mb={4}
        data-test="agreed"
        {...register('agreed')}
        colorScheme="gray">
        <Trans id="TRISA is a network of trusted members. I acknowledge that the contact is authorized to access the organization’s TRISA account information.">
          TRISA is a network of trusted members. I acknowledge that the contact is authorized to
          access the organization’s TRISA account information.
        </Trans>
      </CheckboxFormControl>

      {!isProdEnv ? <DevTool control={control} /> : null}
      <Stack py={5}>
        <InputFormControl
          controlId="name"
          isInvalid={!!errors.name}
          data-test="name"
          formHelperText={errors.name?.message}
          {...register('name')}
          label={
            <>
              <chakra.span fontWeight={700}>
                <Trans id="VASP Name">VASP Name</Trans>
              </chakra.span>{' '}
              (<Trans id="required">required</Trans>)
            </>
          }
        />
      </Stack>
      <InputFormControl
        controlId="email"
        data-test="email"
        isInvalid={!!errors.email}
        formHelperText={errors?.email?.message}
        {...register('email')}
        label={
          <>
            <chakra.span fontWeight={700}>
              <Trans id="Email Address">Email Address</Trans>
            </chakra.span>{' '}
            (<Trans id="required">required</Trans>)
          </>
        }
      />

      <Stack display="flex" flexDir="column" gap={3} py={5}>
        <Button
          bg="orange"
          _hover={{ bg: 'orange' }}
          data-test="submit"
          minW="150px"
          type="submit"
          isDisabled={formValues().agreed === false}
          isLoading={isCreating}>
          Invite
        </Button>
        <Button
          variant="outline"
          mb={4}
          color="link"
          data-test="cancel"
          disabled={isCreating}
          fontWeight={400}
          onClick={onCloseModal}
          minW="150px">
          Cancel
        </Button>
      </Stack>
    </chakra.form>
  );
};

export default AddCollaboratorForm;
