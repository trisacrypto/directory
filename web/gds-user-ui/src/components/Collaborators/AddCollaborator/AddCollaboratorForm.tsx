import { FC, useEffect, useState } from 'react';
import { useForm } from 'react-hook-form';
import { ADD_COLLABORATOR_FORM_METHOD } from './AddCollaboratorFormValidation';
import type { NewCollaborator } from './AddCollaboratorType';
import { useCreateCollaborator } from 'hooks/useCreateCollaborator';
import { Button, Stack, Text, VStack, chakra, useToast } from '@chakra-ui/react';
import { DevTool } from '@hookform/devtools';
import { isProdEnv } from 'application/config';
import InputFormControl from 'components/ui/InputFormControl';
import CustomToast from 'components/ui/CustomToast';
import CheckboxFormControl from 'components/ui/CheckboxFormControl';
import { t, Trans } from '@lingui/macro';
import { useFetchCollaborators } from 'components/Collaborators/useFetchCollaborator';
import { useSelector } from 'react-redux';
import { userSelector } from 'application/store/selectors/user';
type Props = {
  onCloseModal: () => void;
};

const AddCollaboratorForm: FC<Props> = (props) => {
  const { onCloseModal } = props;
  const mutation = useCreateCollaborator();
  const toast = useToast();
  const { getAllCollaborators } = useFetchCollaborators();
  const { user } = useSelector(userSelector);
  const {
    createCollaborator,
    isCreating,
    wasCollaboratorCreated,
    hasCollaboratorFailed,
    errorMessage
  } = mutation;
  const {
    register,
    control,
    handleSubmit,
    formState: { errors }
  } = useForm<NewCollaborator>(ADD_COLLABORATOR_FORM_METHOD) as any;
  const [isChecked, setIsChecked] = useState(false);

  useEffect(() => {
    if (wasCollaboratorCreated) {
      getAllCollaborators();
    }
  }, [wasCollaboratorCreated, getAllCollaborators]);

  useEffect(() => {
    if (wasCollaboratorCreated) {
      onCloseModal();
      toast({
        position: 'top-right',
        title: t`Collaborator has been added successfully`,
        isClosable: true,
        duration: 9000,
        status: 'success'
      });
    }
  }, [wasCollaboratorCreated, onCloseModal, toast]);

  const onSubmit = (data: NewCollaborator) => {
    createCollaborator(data);
    // set collaborators in global state
  };

  if (hasCollaboratorFailed) {
    return (
      <Stack data-test="collaborator-failed">
        <CustomToast
          title="Fail to add a new collaborator"
          description={errorMessage}
          status="error"
        />
      </Stack>
    );
  }

  return (
    <chakra.form onSubmit={handleSubmit(onSubmit)}>
      <VStack mb={5}>
        <Text fontWeight={'bold'}>
          <Trans>Please provide the name and email address for the new contact</Trans>
        </Text>
        <Text fontSize="sm">
          <Trans>
            The contact will receive an email to create a TRISA Global Directory Service (GDS)
            account or join this organization if the contact already has a TRISA GDS account. The
            invitation to join is valid for 7 calendar days. The contact will be added as a member
            for the VASP. The contact will have the ability to contribute to certificate requests,
            check on the status of certificate requests, and complete other actions related to the
            organization’s TRISA membership.
          </Trans>
        </Text>
      </VStack>

      {!isProdEnv ? <DevTool control={control} /> : null}
      <Stack>
        <Text fontWeight={'bold'} size={'md'}>
          <Trans>VASP</Trans>
        </Text>
        <Text>{user?.vasp?.name}</Text>
      </Stack>

      <Stack py={5}>
        <InputFormControl
          controlId="name"
          isInvalid={!!errors.name}
          data-test="name"
          formHelperText={errors.name?.message}
          {...register('name')}
          label={
            <>
              <chakra.span fontWeight={700}>
                <Trans>Contact Name</Trans>
              </chakra.span>{' '}
              (<Trans>required</Trans>)
            </>
          }
        />
      </Stack>
      <InputFormControl
        controlId="email"
        data-test="email"
        isInvalid={!!errors.email}
        formHelperText={errors?.email?.message}
        {...register('email')}
        label={
          <>
            <chakra.span fontWeight={700}>
              <Trans>Email Address</Trans>
            </chakra.span>{' '}
            (<Trans id="required">required</Trans>)
          </>
        }
      />

      <CheckboxFormControl
        controlId="agreed"
        mt={2}
        mb={4}
        {...register('agreed')}
        onChange={(e) => setIsChecked(e.target.checked)}
        size="md"
        borderColor={'black'}
        colorScheme="gray">
        <Trans>
          TRISA is a network of trusted members. I acknowledge that the contact is authorized to
          access the organization’s TRISA account information.
        </Trans>
      </CheckboxFormControl>

      <Stack display="flex" flexDir="column" gap={3} py={5}>
        <Button
          bg="orange"
          _hover={{ bg: 'orange' }}
          data-test="submit"
          minW="150px"
          type="submit"
          isDisabled={!isChecked}
          isLoading={isCreating}>
          Invite
        </Button>
        <Button
          variant="outline"
          mb={4}
          color="link"
          data-test="cancel"
          disabled={isCreating}
          fontWeight={400}
          onClick={onCloseModal}
          minW="150px">
          Cancel
        </Button>
      </Stack>
    </chakra.form>
  );
};

export default AddCollaboratorForm;
