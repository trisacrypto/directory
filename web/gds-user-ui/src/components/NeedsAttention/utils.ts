export const enum AttentionAction {
  'VERIFY_EMAILS' = 'VERIFY_EMAILS',
  'VERIFY_PHONES' = 'VERIFY_PHONES',
  'VERIFY_ADDRESSES' = 'VERIFY_ADDRESSES',
  'VERIFY_IDENTITIES' = 'VERIFY_IDENTITIES',
  'VERIFY_DOCUMENTS' = 'VERIFY_DOCUMENTS',
  'VERIFY_ALL' = 'VERIFY_ALL',
  'VERIFY_NONE' = 'VERIFY_NONE'
}
