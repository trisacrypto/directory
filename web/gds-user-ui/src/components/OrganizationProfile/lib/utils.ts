export const StatusCheck = (status: string) => {
  switch (status) {
    case 'NO_VERIFICATION':
      return 'Not Verified';
    case 'PENDING_REVIEW':
      return 'Pending Review';
    case 'VERIFIED':
      return 'Verified';
    case 'REJECTED':
      return 'Rejected';
    default:
      return 'Pending Registration';
  }
};
