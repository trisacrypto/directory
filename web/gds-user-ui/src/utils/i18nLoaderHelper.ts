import { i18n } from '@lingui/core';
import { en, fr, de, zh, ja, ru } from 'make-plural/plurals';
import { Locales } from 'types/type';

export const DEFAULT_LOCALE: Locales = 'en';

export async function dynamicActivate(locale: string) {
  const { messages } = await import(`../locales/${locale}/messages`);

  i18n.load(locale, messages);
  i18n.loadLocaleData({
    en: { plurals: en },
    de: { plurals: de },
    fr: { plurals: fr },
    zh: { plurals: zh },
    ja: { plurals: ja }
  });
  i18n.activate(locale);
}
