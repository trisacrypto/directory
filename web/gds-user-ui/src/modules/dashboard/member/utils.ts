import { DirectoryType, VaspDirectoryEnum, DirectoryTypeEnum } from './memberType';
import { convertToCVS, downloadCSV } from 'utils/utils';
import { t } from '@lingui/macro';

export const memberTableHeader = [
  {
    key: 'name',
    label: t`Member Name`
  },
  {
    key: 'first_listed',
    label: t`Joined`
  },
  {
    key: 'last_updated',
    label: t`Last Updated`
  },
  {
    key: 'registered_directory',
    label: t`Network`
  },
  {
    key: 'status',
    label: t`Status`
  }
];
export const getVaspDirectory = (dir: DirectoryType) => {
  return dir === DirectoryTypeEnum.TESTNET ? VaspDirectoryEnum.TESTNET : VaspDirectoryEnum.MAINNET;
};

export const getVaspNetwork = (dir: any) => {
  switch (dir) {
    case (VaspDirectoryEnum.TESTNET, VaspDirectoryEnum.TESTNET_DEV):
      return 'TestNet';
    case (VaspDirectoryEnum.MAINNET, VaspDirectoryEnum.MAINNET_DEV):
      return 'MainNet';
    default:
      return 'MainNet';
  }
};

export const getVaspStatus = (status: number) => {
  switch (status) {
    case 1:
      return t`NO VERIFICATION`;
    case 2:
      return t`SUBMITTED`;
    case 3:
      return t`EMAIL VERIFIED`;
    case 4:
      return t`PENDING REVIEW`;
    case 5:
      return t`REVIEWED`;
    case 6:
      return t`VERIFIED`;
    case 7:
      return t`REJECTED`;
    case 8:
      return t`APPEALED`;
    case 9:
      return t`ERRORED`;
    default:
      return t`NO VERIFICATION`;
  }
};

export const downloadMembers2CVS = (member: any) => {
  const m = member.map((item: any) => {
    return {
      ...item,
      status: getVaspStatus(item.status),
      registered_directory: getVaspNetwork(item.registered_directory)
    };
  });
  const memberCsv = convertToCVS(m, memberTableHeader as ITableHeader[]);
  downloadCSV(memberCsv, 'members');
};

