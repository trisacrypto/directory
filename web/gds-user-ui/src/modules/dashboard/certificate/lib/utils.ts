import { basicDetailsValidationSchema } from 'modules/dashboard/certificate/lib/basicDetailsValidationSchema';
import { legalPersonValidationSchemam } from 'modules/dashboard/certificate/lib/legalPersonValidationSchema';
import { contactsValidationSchema } from 'modules/dashboard/certificate/lib/contactsValidationSchema';
import { trisaImplementationValidationSchema } from 'modules/dashboard/certificate/lib/trisaImplementationValidationSchema';
import { trixoQuestionnaireValidationSchema } from 'modules/dashboard/certificate/lib/trixoQuestionnaireValidationSchema';

export const isBasicDetailsCompleted = async (data: any) => {
  try {
    const r = await basicDetailsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    return false;
  }
};

export const isLegalPersonCompleted = async (data: any) => {
  try {
    const r = await legalPersonValidationSchemam.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('legal person validation failed', err);
    return false;
  }
};

export const isContactsCompleted = async (data: any) => {
  try {
    const r = await contactsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('contacts validation failed', err);
    return false;
  }
};

export const isTrisaImplementationCompleted = async (data: any) => {
  try {
    await trisaImplementationValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};

export const isTrixoQuestionnaireCompleted = async (data: any) => {
  try {
    await trixoQuestionnaireValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};
