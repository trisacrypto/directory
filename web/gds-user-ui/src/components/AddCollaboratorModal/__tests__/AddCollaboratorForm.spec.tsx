/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import { waitFor } from '@testing-library/react';
import { useQuery } from '@tanstack/react-query';
import userEvent from '@testing-library/user-event';
import { dynamicActivate } from 'utils/i18nLoaderHelper';
import AddCollaboratorForm from '../AddCollaboratorForm';
import { act, render } from 'utils/test-utils';

function renderComponent() {
  const Props = {
    onCloseModal: jest.fn()
  };

  return render(<AddCollaboratorForm {...Props} />);
}

describe('AddCollaboratorForm', () => {
  beforeAll(() => {
    act(() => {
      dynamicActivate('en');
    });
  });
  it('should render', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
