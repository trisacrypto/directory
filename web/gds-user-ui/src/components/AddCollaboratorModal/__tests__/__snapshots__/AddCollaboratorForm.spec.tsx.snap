// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCollaboratorForm should render 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-4 {
  width: 100%;
  position: relative;
}

.emotion-5 {
  display: block;
  text-align: start;
  font-size: var(--chakra-fontSizes-md);
  -webkit-margin-end: var(--chakra-space-3);
  margin-inline-end: var(--chakra-space-3);
  margin-bottom: var(--chakra-space-2);
  font-weight: var(--chakra-fontWeights-medium);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  opacity: 1;
}

.emotion-5[disabled],
.emotion-5[aria-disabled=true],
.emotion-5[data-disabled] {
  opacity: 0.4;
}

.emotion-6 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  position: relative;
  margin-top: var(--chakra-space-2);
  margin-bottom: var(--chakra-space-4);
}

.emotion-6[disabled],
.emotion-6[aria-disabled=true],
.emotion-6[data-disabled] {
  cursor: not-allowed;
}

.emotion-7 {
  width: var(--chakra-sizes-4);
  transition-property: box-shadow;
  transition-duration: var(--chakra-transition-duration-normal);
  border: 2px solid;
  border-radius: var(--chakra-radii-sm);
  border-color: inherit;
  color: var(--chakra-colors-white);
  height: var(--chakra-sizes-4);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: top;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-7[aria-checked=true],
.emotion-7[data-checked] {
  background: var(--chakra-colors-gray-500);
  border-color: var(--chakra-colors-gray-500);
  color: var(--chakra-colors-white);
}

.emotion-7[aria-checked=true]:hover,
.emotion-7[data-checked]:hover,
.emotion-7[aria-checked=true][data-hover],
.emotion-7[data-checked][data-hover] {
  background: var(--chakra-colors-gray-600);
  border-color: var(--chakra-colors-gray-600);
}

.emotion-7[aria-checked=true][disabled],
.emotion-7[data-checked][disabled],
.emotion-7[aria-checked=true][aria-disabled=true],
.emotion-7[data-checked][aria-disabled=true],
.emotion-7[aria-checked=true][data-disabled],
.emotion-7[data-checked][data-disabled] {
  border-color: var(--chakra-colors-gray-200);
  background: var(--chakra-colors-gray-200);
  color: var(--chakra-colors-gray-500);
}

.emotion-7:indeterminate,
.emotion-7[aria-checked=mixed],
.emotion-7[data-indeterminate] {
  background: var(--chakra-colors-gray-500);
  border-color: var(--chakra-colors-gray-500);
  color: var(--chakra-colors-white);
}

.emotion-7[disabled],
.emotion-7[aria-disabled=true],
.emotion-7[data-disabled] {
  background: var(--chakra-colors-gray-100);
  border-color: var(--chakra-colors-gray-100);
}

.emotion-7:focus,
.emotion-7[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-7[aria-invalid=true],
.emotion-7[data-invalid] {
  border-color: var(--chakra-colors-red-500);
}

.emotion-8 {
  -webkit-margin-start: 0.5rem;
  margin-inline-start: 0.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: var(--chakra-fontSizes-md);
}

.emotion-8[disabled],
.emotion-8[aria-disabled=true],
.emotion-8[data-disabled] {
  opacity: 0.4;
}

.emotion-9 {
  margin-top: var(--chakra-space-2);
  color: var(--chakra-colors-gray-500);
  line-height: var(--chakra-lineHeights-normal);
  font-size: var(--chakra-fontSizes-sm);
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: var(--chakra-space-5);
  padding-bottom: var(--chakra-space-5);
}

.emotion-10>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-13 {
  font-weight: 700;
}

.emotion-14 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.emotion-15 {
  width: 100%;
  min-width: 0px;
  outline: 2px solid transparent;
  outline-offset: 2px;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  height: var(--chakra-sizes-10);
  border-radius: 0px;
  border: 1px solid;
  border-color: inherit;
  background: #E3EBEF;
}

.emotion-15:hover,
.emotion-15[data-hover] {
  border-color: var(--chakra-colors-gray-300);
}

.emotion-15[aria-readonly=true],
.emotion-15[readonly],
.emotion-15[data-readonly] {
  box-shadow: none!important;
  -webkit-user-select: all;
  -moz-user-select: all;
  -ms-user-select: all;
  user-select: all;
}

.emotion-15[disabled],
.emotion-15[aria-disabled=true],
.emotion-15[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
}

.emotion-15[aria-invalid=true],
.emotion-15[data-invalid] {
  border-color: #E53E3E;
  box-shadow: 0 0 0 1px #E53E3E;
}

.emotion-15:focus,
.emotion-15[data-focus] {
  z-index: 1;
  border-color: #3182ce;
  box-shadow: 0 0 0 1px #3182ce;
}

.emotion-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: var(--chakra-space-3);
  padding-top: var(--chakra-space-5);
  padding-bottom: var(--chakra-space-5);
}

.emotion-23>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-24 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: 150px;
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: orange;
}

.emotion-24:focus,
.emotion-24[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-24[disabled],
.emotion-24[aria-disabled=true],
.emotion-24[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-24:hover,
.emotion-24[data-hover] {
  background: orange;
}

.emotion-24:active,
.emotion-24[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-25 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: 400;
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: 150px;
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  border: 1px solid;
  border-color: var(--chakra-colors-gray-200);
  color: link;
  margin-bottom: var(--chakra-space-4);
}

.emotion-25:focus,
.emotion-25[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-25[disabled],
.emotion-25[aria-disabled=true],
.emotion-25[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-25:hover,
.emotion-25[data-hover] {
  background: var(--chakra-colors-gray-100);
}

.emotion-25:hover[disabled],
.emotion-25[data-hover][disabled],
.emotion-25:hover[aria-disabled=true],
.emotion-25[data-hover][aria-disabled=true],
.emotion-25:hover[data-disabled],
.emotion-25[data-hover][data-disabled] {
  background: initial;
}

.emotion-25:active,
.emotion-25[data-active] {
  background: var(--chakra-colors-gray-200);
}

<div>
  <form
    class="emotion-0"
  >
    <div
      class="chakra-stack emotion-1"
    >
      <p
        class="chakra-text emotion-0"
      >
        Please provide the name of the VASP and email address for the new contact.
      </p>
      <p
        class="chakra-text emotion-0"
      >
        The contact will receive an email to create a TRISA Global Directory Service Account. The invitation to join is valid for 7 calendar days. The contact will be added as a member for the VASP. The contact will have the ability to contribute to certificate requests, check on the status of certificate requests, and complete other actions related to the organization’s TRISA membership.
      </p>
    </div>
    <div
      class="chakra-form-control emotion-4"
      role="group"
    >
      <label
        class="chakra-form__label emotion-5"
        for="agreed"
        id="field-1-label"
      >
        agreed
      </label>
      <label
        class="chakra-checkbox emotion-6"
        data-test="agreed"
      >
        <input
          aria-describedby="field-1-helptext"
          aria-disabled="false"
          aria-invalid="false"
          class="chakra-checkbox__input"
          id="agreed"
          name="agreed"
          style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
          type="checkbox"
          value=""
        />
        <span
          aria-hidden="true"
          class="chakra-checkbox__control emotion-7"
        />
        <span
          class="chakra-checkbox__label emotion-8"
        >
          TRISA is a network of trusted members. I acknowledge that the contact is authorized to access the organization’s TRISA account information.
        </span>
      </label>
      <div
        class="chakra-form__helper-text emotion-9"
        id="field-1-helptext"
      />
    </div>
    <div
      class="chakra-stack emotion-10"
    >
      <div
        class="chakra-form-control emotion-4"
        role="group"
      >
        <label
          class="chakra-form__label emotion-5"
          for="name"
          id="field-2-label"
        >
          <span
            class="emotion-13"
          >
            VASP Name
          </span>
           
          (
          required
          )
        </label>
        <div
          class="chakra-input__group emotion-14"
        >
          <input
            aria-describedby="field-2-helptext"
            class="chakra-input emotion-15"
            data-test="name"
            id="name"
            name="name"
            type="text"
          />
        </div>
        <div
          class="chakra-form__helper-text emotion-9"
          id="field-2-helptext"
        />
      </div>
    </div>
    <div
      class="chakra-form-control emotion-4"
      role="group"
    >
      <label
        class="chakra-form__label emotion-5"
        for="email"
        id="field-3-label"
      >
        <span
          class="emotion-13"
        >
          Email Address
        </span>
         
        (
        required
        )
      </label>
      <div
        class="chakra-input__group emotion-14"
      >
        <input
          aria-describedby="field-3-helptext"
          class="chakra-input emotion-15"
          data-test="email"
          id="email"
          name="email"
          type="text"
        />
      </div>
      <div
        class="chakra-form__helper-text emotion-9"
        id="field-3-helptext"
      />
    </div>
    <div
      class="chakra-stack emotion-23"
    >
      <button
        class="chakra-button emotion-24"
        data-test="submit"
        disabled=""
        type="submit"
      >
        Invite
      </button>
      <button
        class="chakra-button emotion-25"
        data-test="cancel"
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
